arangod/Cluster/FollowerInfo.cpp

```diff
@@ -246,46 +153,191 @@ Result FollowerInfo::remove(ServerID const& sid) {
+  // Reset local structures, if we take over leadership we do not know anything!
+  _followers = std::make_shared<std::vector<ServerID>>();
+  _failoverCandidates = std::make_shared<std::vector<ServerID>>();
+  // We disallow writes until the first write.
+  _canWrite = false;
+  // Take over leadership
+  _theLeader = "";
+  _theLeaderTouched = true;
+  TRI_ASSERT(_failoverCandidates != nullptr && _failoverCandidates->empty());
+  if (previousInsyncFollowers.size() > 1) {
+    auto ourselves = arangodb::ServerState::instance()->getId();
+    auto failoverCandidates =
+        std::make_shared<std::vector<ServerID>>(previousInsyncFollowers);
+    auto myEntry =
+        std::find(failoverCandidates->begin(), failoverCandidates->end(), ourselves);
+    // We are a valid failover follower
+    TRI_ASSERT(myEntry != failoverCandidates->end());
+    // The first server is a different leader! (For some reason the job can be
+    // triggered twice) TRI_ASSERT(myEntry != failoverCandidates->begin());
+    failoverCandidates->erase(myEntry);
+    // Put us in front, put old leader somewhere, we do not really care
+    _failoverCandidates = failoverCandidates;
+  }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// @brief Update the current information in the Agency. We update the failover-
+///        list with the newest values, after this the guarantee is that
+///        _followers == _failoverCandidates
+////////////////////////////////////////////////////////////////////////////////
+bool FollowerInfo::updateFailoverCandidates() {
+  MUTEX_LOCKER(agencyLocker, _agencyMutex);
+  // Acquire _canWriteLock first
+  WRITE_LOCKER(canWriteLocker, _canWriteLock);
+  // Next acquire _dataLock
+  WRITE_LOCKER(dataLocker, _dataLock);
+  if (_canWrite) {
+// Short circuit, we have multiple writes in the above write lock
+// The first needs to do things and flips _canWrite
+// All followers can return as soon as the lock is released
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+    TRI_ASSERT(_failoverCandidates->size() == _followers->size());
+    std::vector<string> diff;
+    std::set_intersection(_failoverCandidates->begin(),
+                          _failoverCandidates->end(), _followers->begin(),
+                          _followers->end(), std::back_inserter(diff));
+    TRI_ASSERT(diff.empty());
+#endif
+    return _canWrite;
+  }
+  TRI_ASSERT(_followers->size() + 1 >= _docColl->minReplicationFactor());
+  // Update both lists (we use a copy here, as we are modifying them in other places individually!)
+  _failoverCandidates = std::make_shared<std::vector<ServerID> const>(*_followers);
+  // Just be sure
+  TRI_ASSERT(_failoverCandidates.get() != _followers.get());
+  TRI_ASSERT(_failoverCandidates->size() == _followers->size());
+#ifdef ARANGODB_ENABLE_MAINTAINER_MODE
+  std::vector<string> diff;
+  std::set_intersection(_failoverCandidates->begin(), _failoverCandidates->end(),
```

Comment:
As above, did you mean to write
```suggestion
  std::set_symmetric_difference(_failoverCandidates->begin(), _failoverCandidates->end(),
```
?

Reply:
Right thanks!